From ba8e5efdf5a8bb2d3dd2b20c751ea8ad94869492 Mon Sep 17 00:00:00 2001
From: Viktor Rehnberg <viktor.rehnberg@gmail.com>
Date: Wed, 25 Sep 2024 09:29:23 +0000
Subject: [PATCH] Add LD_LIBRARY_PATH and PATH to env vars

The multinode runner launches processes with pdsh, if LD_LIBRARY_PATH is
not included in these exports then the python .so file may not be found.

See https://github.com/easybuilders/easybuild-easyconfigs/pull/21438#issuecomment-2373540098
for more details.
---
 deepspeed/launcher/runner.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deepspeed/launcher/runner.py b/deepspeed/launcher/runner.py
index 07d1713e..498cd829 100755
--- a/deepspeed/launcher/runner.py
+++ b/deepspeed/launcher/runner.py
@@ -31,7 +31,7 @@ from ..autotuning import Autotuner
 from deepspeed.accelerator import get_accelerator
 
 DLTS_HOSTFILE = "/job/hostfile"
-EXPORT_ENVS = ['MLFLOW', 'PYTHON', 'MV2', 'UCX']
+EXPORT_ENVS = ['MLFLOW', 'PYTHON', 'MV2', 'UCX', 'LD_LIBRARY_PATH', 'PATH']
 EXPORT_ENVS += NEBULA_EXPORT_ENVS
 DEEPSPEED_ENVIRONMENT_NAME = os.getenv("DS_ENV_FILE", ".deepspeed_env")
 DEEPSPEED_ENVIRONMENT_PATHS = [os.path.expanduser("~"), '.']
-- 
2.39.3

